dipy/denoise/denspeed.pyx

```diff
@@ -36,7 +36,7 @@ def nlmeans_3d(arr, mask=None, sigma=None, patch_radius=1,
     """
 
     if arr.ndim != 3:
-        raise ValueError('arr needs to be a 3D ndarray')
+        raise ValueError('data needs to be a 3D ndarray', arr.shape)
```

Comment:
No need for replacing arr with data here.  Except if you change arr with data in the parameters. But if you do that then we will need to writing down in API  changes.